### ISSUE TITLE
Table of Contents Component

### ISSUE THREAD

**Issue by rgbkrk** [nteract/archived-desktop-app#1496]
Having a table of contents we can view somewhere would be excellent. I'd like the component itself to be able to take the notebook state and handle clicking the headers in the ToC.

```js
<NotebookTableOfContents
  notebook={notebook}
  onClick={(cellID) => { console.log(cellID) } }
/>
```

MiniMap would be cool too. I think these components could be started off as a new package and iterated on using `notebook-preview-demo`.

**Comment by captainsafia** [comment]
I think I can work on this (although not sure how soon).

I assume by MiniMap you mean something like Sublime Text's source code map that appears on the right?

**Comment by rgbkrk** [comment]
Yeah the little raster looking thing that is on the left or right side. https://i.stack.imgur.com/FNfcn.jpg

**Comment by rgbkrk** [comment]
Note: that's not my image I just grabbed one from Google image search that had arrows. :smile:

**Comment by captainsafia** [comment]
I guess you really can find anything on the Internet!

**Comment by alanzchen** [comment]
I would definitely vote for this feature!

**Comment by spencerbertsch1** [comment]
Any updates on this feature? Would love to see it come to nteract!

**Comment by captainsafia** [comment]
I haven't had the chance to work on this and I'm not sure that anyone else has, so it's still open for the moment. I'm open to helping someone else work on submitting a pull request for this.

**Comment by captainsafia** [comment]
We should be able to tackle this soon with our upcoming UI overhaul.

**Comment by captainsafia** [comment]
Pinning this since we want to pursue this work at some point.
